### PR TITLE
Support delayed functions

### LINKED ISSUE #1: How can I use with dask.delayed functions?
Could You provide an example to map functions returning `Delayed` objects?

### PR DESCRIPTION
**Pull request by kszucs**
Resolves #1

### CODE REVIEW

File: streams/tests/test_dask.py [hunk 1]
```diff
@@ -54,6 +55,33 @@ def test_map(c, s, a, b):
...
+    def inc1(x):
+        """Immediate delayed function"""
+        return inc(x)
+
+    def inc2(x):
+        """Function that just returns a delayed"""
+        return delayed(inc)(x)
+
+    L1 = source.to_dask().compute(inc1).sink_to_list()
+    L2 = source.to_dask().compute(inc2).sink_to_list()
```
Comment (mrocklin) [inline streams/tests/test_dask.py]: What is your objective here?  What are you trying to accomplish?  Is this preferable to `.map(inc)` in some way?

Comment (kszucs) [inline streams/tests/test_dask.py]: I have an already implemented function which creates the following dag from an input record:

image

Theoretically I could create the same `streaming DAG` but just computing a delayed is easier to start with.

Comment (kszucs) [inline streams/tests/test_dask.py]: Do You mean placing the same logic under `map`? That was my first try, but I thought this is more explicit. `Client.submit` doesn't handle delayed objects, I need to use `Client.compute` instead.

Comment (mrocklin) [inline streams/tests/test_dask.py]: Just thinking aloud, but thoughts on a function that could take delayed values as inputs and produce the equivalent streaming DAG?

    stream_inputs, stream_outputs = make_stream(delayed_inputs, delayed_outputs)

Comment (kszucs) [inline streams/tests/test_dask.py]: Wouldn't it hurt the distributed scheduler? Seems reasonable and complex to implement :)

Comment (mrocklin) [inline streams/tests/test_dask.py]: Hurt it how?

I agree that this is more complex and that we shouldn't do it if not useful.  I'm thinking about cases where your inputs and outputs come in and leave at different rates.  It looks like your graph has four inputs.  How do you apply this graph onto one element coming through the stream?  It is a tuple of four things maybe?

Also, just to be clear I'm not negative on your PR, just thinking about other alternatives in this space.

Comment (kszucs) [inline streams/tests/test_dask.py]: Hurt the Performance of the scheduler, never mind!

**Comment by kszucs** [comment]
The input is a dictionary containing 4 keys in this particular case.

```python
def pipeline(repos, account_id, project_id, connection):
    # clone repos and preprocess extracted dataframes
    frames = valmap(clone, repos)  # clone is a delayed function
    frames = valmap(preprocess, frames)  # preprocess is a delayed function

    # extract commit and patchframes
    cdfs = valmap(get(0), frames)  # toolz.curried.get
    pdfs = valmap(get(1), frames)

    # create authorframe with deduplication rules
    cdf = delayed(pd.concat)(cdfs, names=['repo_id'])
    cdf, adf = deduplicate(cdf)  # delayed 

    # hour and effort estimation
    cdf = estimate_hours(cdf)  # delayed

    ncommits = persist_commits(cdf, account_id, project_id, connection)
    nauthors = persist_authors(adf, account_id, project_id, connection)
    npatches = [persist_patches(pdf, account_id, project_id, repo_id,
                                connection)
                for repo_id, pdf in pdfs.items()]
    npatches = delayed(sum)(npatches)

    return ncommits, npatches, nauthors
```

**Comment by CJ-Wright** [comment]
@kszucs Does the current setup with https://github.com/mrocklin/streamz/blob/master/streamz/dask.py support what you are asking for? If so can you please close?

**Pull request closed without merging**